pkg/controller/install/install_controller.go

```diff
@@ -108,12 +109,73 @@ func (r *ReconcileInstall) Reconcile(request reconcile.Request) (reconcile.Resul
 	return reconcile.Result{}, nil
 }
 
+// This is a transform method that ignores clusterrole and clusterrolebinding
+// resources for namespace scoped deployment of kubefed-operator
+func resourceScopeFilter(scope servingv1alpha1.InstallationScope) mf.Transformer {
+	return func(u *unstructured.Unstructured) *unstructured.Unstructured {
+		if scope == servingv1alpha1.InstallationScopeNamespaceScoped {
+			switch strings.ToLower(u.GetKind()) {
+			case "clusterrole":
+			case "clusterrolebinding":
+				return nil
+			}
+		}
+		return u
+	}
+}
+
+// This is a transform method that updates the deployment resource's environment variables
+// by adding the federation scope env. variable for namespace scoped deployments
+func resourceEnvUpdate(scope servingv1alpha1.InstallationScope) mf.Transformer {
+	return func(u *unstructured.Unstructured) *unstructured.Unstructured {
+		if scope == servingv1alpha1.InstallationScopeNamespaceScoped {
+			switch strings.ToLower(u.GetKind()) {
+			case "deployment":
+				if envs, ok, err := unstructured.NestedSlice(u.Object,
+					"spec", "template", "spec", "containers[0]", "env"); ok {
+					fse := map[string]string{"name": "DEFAULT_FEDERATION_SCOPE", "value": "Namespaced"}
+					envs = append(envs, fse)
+					err = unstructured.SetNestedSlice(u.Object, envs, "spec",
+						"template", "spec", "containers[0]", "env")
+					if err != nil {
+						reqLogger := log.WithValues("Instance.Namespace", u.GetNamespace(), "Instance.Name", u.GetName())
+
+						reqLogger.Info("Failed to update the environment")
+					}
+				}
+			}
+		}
+		return u
+	}
+}
+
+// This is a transform method that updates the namespace field of the clusterrolebinding resource
+// for cluster scoped deployment
+func resourceNamespaceUpdate(scope servingv1alpha1.InstallationScope, ns string) mf.Transformer {
+	return func(u *unstructured.Unstructured) *unstructured.Unstructured {
+		if scope == servingv1alpha1.InstallationScopeClusterScoped {
+			switch strings.ToLower(u.GetKind()) {
+			case "clusterrolebinding":
+				err := unstructured.SetNestedField(u.Object, ns, "subjects[0]", "namespace")
+				if err != nil {
+					reqLogger := log.WithValues("Instance.Namespace", ns, "Instance.Name", u.GetName())
```

Comment:
u.GetName() is the name of the unstructured, not the name of the Instance

Reply:
Do you think it's worth passing the name of the instance as an additional argument for correctness of the logger?

Reply:
I could take it or leave it, since we'll be changing this all around anyway with the new version of manifestival.